## Name :- Varun Biren Majithia
## Student_id :- 8894799

### Importing torch

In [ ]:
import torch

### Calculate the dot product between two vectors each of length 3 using pytorch

In [2]:
# creating vectors v1 and v2 of length 3
v1  = torch.tensor([16,17,18])
v2 = torch.tensor([19,20,21])

# Calculate the dot product
dot_prdt = torch.dot(v1,v2)

print(dot_prdt)


tensor(1022)


Inferences:- 

The torch.tensor fucntion is used to create two tensors representing the vectors v1 and v2 respectively.

torch.dot function computes the dot prooduct between two vectors

And the output tensor(1022) represents a scalar tensor


### Calculate the matrix multiplication of two 2-dimensional matrices, each with a size of (2,2) using pytorch

In [3]:
# Creating 2D matrices mat1 and mat2 of size 2x2

mat1 = torch.tensor([[8,9], [10,11]])
mat2 = torch.tensor([[12,13], [11,22]])

# Performing matrix multiplication
output = torch.matmul(mat1,mat2)


print(output)

tensor([[195, 302],
        [241, 372]])


Inferences:- 

The torch.tensor is used to create two tensors representing the matrices.

torch.matmul is used to perform matrix multiplication bterween mat1 and mat2

The final output is a Pytorch tensor representing matrix product

## PART B

### Calculate the dot product between the same vectors as above using tensorflow
### Calculate the matrix multiplication of the same matrics as above using tensorflow

In [4]:
import tensorflow as tf

# Define the two vectors vector1 and vector 2 of lenght 3
vector1 = tf.constant([9,10,11])
vector2= tf.constant([16,11,18])

# Calculate the dot product
dot_prod = tf.tensordot(vector1, vector2, axes=1)

# Print the result
print(dot_prod)

tf.Tensor(452, shape=(), dtype=int32)


Inferences :- 

The tf.constant function is used to create two constant tensors represnting the vectors

To compute the dot product between two vectors we use the tf.tensordot 

Finally, the output is a tensorflow tensor which represents the dot product of the vectors

In [5]:
import tensorflow as tf

# Define the two 2D matrices of shape 2x2 
matrix_1 = tf.constant([[2,3], [4,5]])
matrix_2 = tf.constant([[7,7], [11,12]])

# Perform matrix multiplication
result = tf.matmul(matrix_1, matrix_2)

# Print the result
print(result)

tf.Tensor(
[[47 50]
 [83 88]], shape=(2, 2), dtype=int32)


Inferences :-

the tf.constant is used to create two constant tensors reprensting two matrices.

tf.matmul function is used to perform matrix multiplication.

The output is a tensorflow tensor representing the matrix product 